integration-tests/tests/run-analysis-azure.js

```diff
@@ -0,0 +1,94 @@
+// This test is owned by the Interactive Analysis (IA) Team.
+const _ = require('lodash/fp');
+const uuid = require('uuid');
+const { deleteAppsV2, deleteRuntimesV2, withWorkspaceAzure, performAnalysisTabSetup } = require('../utils/integration-helpers');
+const {
+  click,
+  clickable,
+  delay,
+  findElement,
+  noSpinnersAfter,
+  fillIn,
+  findIframe,
+  findText,
+  dismissNotifications,
+  getAnimatedDrawer,
+  image,
+  input,
+  Millis,
+} = require('../utils/integration-utils');
+const { registerTest } = require('../utils/jest-utils');
+const { withUserToken } = require('../utils/terra-sa-utils');
+
+const notebookName = `test-notebook-${uuid.v4()}`;
+const waitAMinute = { timeout: Millis.ofMinute };
+
+const testRunAnalysisAzure = _.flowRight(
+  withUserToken,
+  withWorkspaceAzure
+)(async ({ billingProject, workspaceName, page, testUrl, token }) => {
+  await performAnalysisTabSetup(page, token, testUrl, workspaceName);
+  await delay(Millis.ofSeconds(10));
```

Comment:
All of these delays of arbitrary times is an anti-pattern in robust browser-based testing. How do you know you have waited long enough (which can lead to flakiness if operations take longer than you expected)? Are you waiting too long (which leads to tests that are longer than you need)? 

Whenever possible, you should poll for some condition to be complete as a signal for the test to move on. Delays should only be used as an absolute last resort, for instance in the case of having to handle an animated modal closing.

Reply:
Thanks, I'd prefer not to use delays, of course! I'm still working on getting this running all the way through; the polling methods like `noSpinnersAfter` and the helpers designed to wait for animations to finish have been giving me a lot of trouble.

Reply:
Thanks for the impetus to inspect our helper methods - managed to excise all `delay`s with new or improved `waitForWhatever` methods <3

Reply:
That's awesome!! Thanks so much for working through that.

When you get tests into a happy passing state, let me know and I will review the full PR.